airflow/models/connection.py

```diff
@@ -128,6 +128,27 @@ def __init__(
 
     def parse_from_uri(self, uri):
         uri_parts = urlparse(uri)
+        if uri_parts.scheme in ['http', 'https']:
+            self._parse_from_http_uri(uri)
+        else:
+            self._parse_from_airflow_uri(uri)
+
+    def _parse_from_http_uri(self, uri):
+        self.conn_type = 'http'
+        url_parts = urlparse(uri)
+        netloc = "%s:%s" % (url_parts.hostname, url_parts.port) if url_parts.port else url_parts.hostname
+        # The fragment is used to store extra configuration. The HTTP client never uses fragments, so
```

Comment:
This needs documenting somewhere that end users will be able to see without finding this bit of the code.

Reply:
Yes. Of course. First, I wanted to determine the implementations.